File: 374/Easy.hs
-- https://www.reddit.com/r/dailyprogrammer/comments/akv6z4/20190128_challenge_374_easy_additive_persistence/

import Data.List
import Data.Maybe

digitSum :: Integer -> Integer
digitSum 0 = 0
digitSum n = n `mod` 10 + digitSum (n `div` 10)

-- naive
additivePersistence :: Integer -> Int
additivePersistence n | n < 10 = 0
                      | otherwise = 1 + additivePersistence (digitSum n)

-- still naive but slightly more interesting
additivePersistence' :: Integer -> Int
additivePersistence' n = fromJust $ findIndex (<10) digitSums
    where digitSums = n : map digitSum digitSums

File: 122/Easy.jl
# https://www.reddit.com/r/dailyprogrammer/comments/1berjh/040113_challenge_122_easy_sum_them_digits/

function digitalRoot(n)
  n >= 10 ? digitalRoot(digitSum(n)) : n
end

function digitSum(n)
  sum = 0
  while n > 0
    sum += n%10
    n = div(n, 10)
  end
  sum
end
